// @livekit/rtc-node@0.13.14/dist/proto/ffi_pb.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var ffi_pb_exports = {};
__export(ffi_pb_exports, {
  DisposeCallback: () => DisposeCallback,
  DisposeRequest: () => DisposeRequest,
  DisposeResponse: () => DisposeResponse,
  FfiEvent: () => FfiEvent,
  FfiRequest: () => FfiRequest,
  FfiResponse: () => FfiResponse,
  LogBatch: () => LogBatch,
  LogLevel: () => LogLevel,
  LogRecord: () => LogRecord,
  Panic: () => Panic
});
module.exports = __toCommonJS(ffi_pb_exports);
var import_protobuf = require("@bufbuild/protobuf");
var import_room_pb = require("./room_pb.cjs");
var import_track_pb = require("./track_pb.cjs");
var import_video_frame_pb = require("./video_frame_pb.cjs");
var import_audio_frame_pb = require("./audio_frame_pb.cjs");
var import_e2ee_pb = require("./e2ee_pb.cjs");
var import_rpc_pb = require("./rpc_pb.cjs");
var import_track_publication_pb = require("./track_publication_pb.cjs");
var LogLevel = /* @__PURE__ */ ((LogLevel2) => {
  LogLevel2[LogLevel2["LOG_ERROR"] = 0] = "LOG_ERROR";
  LogLevel2[LogLevel2["LOG_WARN"] = 1] = "LOG_WARN";
  LogLevel2[LogLevel2["LOG_INFO"] = 2] = "LOG_INFO";
  LogLevel2[LogLevel2["LOG_DEBUG"] = 3] = "LOG_DEBUG";
  LogLevel2[LogLevel2["LOG_TRACE"] = 4] = "LOG_TRACE";
  return LogLevel2;
})(LogLevel || {});
import_protobuf.proto2.util.setEnumType(LogLevel, "livekit.proto.LogLevel", [
  { no: 0, name: "LOG_ERROR" },
  { no: 1, name: "LOG_WARN" },
  { no: 2, name: "LOG_INFO" },
  { no: 3, name: "LOG_DEBUG" },
  { no: 4, name: "LOG_TRACE" }
]);
const _FfiRequest = class _FfiRequest extends import_protobuf.Message {
  constructor(data) {
    super();
    /**
     * @generated from oneof livekit.proto.FfiRequest.message
     */
    this.message = { case: void 0 };
    import_protobuf.proto2.util.initPartial(data, this);
  }
  static fromBinary(bytes, options) {
    return new _FfiRequest().fromBinary(bytes, options);
  }
  static fromJson(jsonValue, options) {
    return new _FfiRequest().fromJson(jsonValue, options);
  }
  static fromJsonString(jsonString, options) {
    return new _FfiRequest().fromJsonString(jsonString, options);
  }
  static equals(a, b) {
    return import_protobuf.proto2.util.equals(_FfiRequest, a, b);
  }
};
_FfiRequest.runtime = import_protobuf.proto2;
_FfiRequest.typeName = "livekit.proto.FfiRequest";
_FfiRequest.fields = import_protobuf.proto2.util.newFieldList(() => [
  { no: 2, name: "dispose", kind: "message", T: DisposeRequest, oneof: "message" },
  { no: 3, name: "connect", kind: "message", T: import_room_pb.ConnectRequest, oneof: "message" },
  { no: 4, name: "disconnect", kind: "message", T: import_room_pb.DisconnectRequest, oneof: "message" },
  { no: 5, name: "publish_track", kind: "message", T: import_room_pb.PublishTrackRequest, oneof: "message" },
  { no: 6, name: "unpublish_track", kind: "message", T: import_room_pb.UnpublishTrackRequest, oneof: "message" },
  { no: 7, name: "publish_data", kind: "message", T: import_room_pb.PublishDataRequest, oneof: "message" },
  { no: 8, name: "set_subscribed", kind: "message", T: import_room_pb.SetSubscribedRequest, oneof: "message" },
  { no: 9, name: "set_local_metadata", kind: "message", T: import_room_pb.SetLocalMetadataRequest, oneof: "message" },
  { no: 10, name: "set_local_name", kind: "message", T: import_room_pb.SetLocalNameRequest, oneof: "message" },
  { no: 11, name: "set_local_attributes", kind: "message", T: import_room_pb.SetLocalAttributesRequest, oneof: "message" },
  { no: 12, name: "get_session_stats", kind: "message", T: import_room_pb.GetSessionStatsRequest, oneof: "message" },
  { no: 13, name: "publish_transcription", kind: "message", T: import_room_pb.PublishTranscriptionRequest, oneof: "message" },
  { no: 14, name: "publish_sip_dtmf", kind: "message", T: import_room_pb.PublishSipDtmfRequest, oneof: "message" },
  { no: 15, name: "create_video_track", kind: "message", T: import_track_pb.CreateVideoTrackRequest, oneof: "message" },
  { no: 16, name: "create_audio_track", kind: "message", T: import_track_pb.CreateAudioTrackRequest, oneof: "message" },
  { no: 17, name: "local_track_mute", kind: "message", T: import_track_pb.LocalTrackMuteRequest, oneof: "message" },
  { no: 18, name: "enable_remote_track", kind: "message", T: import_track_pb.EnableRemoteTrackRequest, oneof: "message" },
  { no: 19, name: "get_stats", kind: "message", T: import_track_pb.GetStatsRequest, oneof: "message" },
  { no: 48, name: "set_track_subscription_permissions", kind: "message", T: import_track_pb.SetTrackSubscriptionPermissionsRequest, oneof: "message" },
  { no: 20, name: "new_video_stream", kind: "message", T: import_video_frame_pb.NewVideoStreamRequest, oneof: "message" },
  { no: 21, name: "new_video_source", kind: "message", T: import_video_frame_pb.NewVideoSourceRequest, oneof: "message" },
  { no: 22, name: "capture_video_frame", kind: "message", T: import_video_frame_pb.CaptureVideoFrameRequest, oneof: "message" },
  { no: 23, name: "video_convert", kind: "message", T: import_video_frame_pb.VideoConvertRequest, oneof: "message" },
  { no: 24, name: "video_stream_from_participant", kind: "message", T: import_video_frame_pb.VideoStreamFromParticipantRequest, oneof: "message" },
  { no: 25, name: "new_audio_stream", kind: "message", T: import_audio_frame_pb.NewAudioStreamRequest, oneof: "message" },
  { no: 26, name: "new_audio_source", kind: "message", T: import_audio_frame_pb.NewAudioSourceRequest, oneof: "message" },
  { no: 27, name: "capture_audio_frame", kind: "message", T: import_audio_frame_pb.CaptureAudioFrameRequest, oneof: "message" },
  { no: 28, name: "clear_audio_buffer", kind: "message", T: import_audio_frame_pb.ClearAudioBufferRequest, oneof: "message" },
  { no: 29, name: "new_audio_resampler", kind: "message", T: import_audio_frame_pb.NewAudioResamplerRequest, oneof: "message" },
  { no: 30, name: "remix_and_resample", kind: "message", T: import_audio_frame_pb.RemixAndResampleRequest, oneof: "message" },
  { no: 31, name: "e2ee", kind: "message", T: import_e2ee_pb.E2eeRequest, oneof: "message" },
  { no: 32, name: "audio_stream_from_participant", kind: "message", T: import_audio_frame_pb.AudioStreamFromParticipantRequest, oneof: "message" },
  { no: 33, name: "new_sox_resampler", kind: "message", T: import_audio_frame_pb.NewSoxResamplerRequest, oneof: "message" },
  { no: 34, name: "push_sox_resampler", kind: "message", T: import_audio_frame_pb.PushSoxResamplerRequest, oneof: "message" },
  { no: 35, name: "flush_sox_resampler", kind: "message", T: import_audio_frame_pb.FlushSoxResamplerRequest, oneof: "message" },
  { no: 36, name: "send_chat_message", kind: "message", T: import_room_pb.SendChatMessageRequest, oneof: "message" },
  { no: 37, name: "edit_chat_message", kind: "message", T: import_room_pb.EditChatMessageRequest, oneof: "message" },
  { no: 38, name: "perform_rpc", kind: "message", T: import_rpc_pb.PerformRpcRequest, oneof: "message" },
  { no: 39, name: "register_rpc_method", kind: "message", T: import_rpc_pb.RegisterRpcMethodRequest, oneof: "message" },
  { no: 40, name: "unregister_rpc_method", kind: "message", T: import_rpc_pb.UnregisterRpcMethodRequest, oneof: "message" },
  { no: 41, name: "rpc_method_invocation_response", kind: "message", T: import_rpc_pb.RpcMethodInvocationResponseRequest, oneof: "message" },
  { no: 42, name: "enable_remote_track_publication", kind: "message", T: import_track_publication_pb.EnableRemoteTrackPublicationRequest, oneof: "message" },
  { no: 43, name: "update_remote_track_publication_dimension", kind: "message", T: import_track_publication_pb.UpdateRemoteTrackPublicationDimensionRequest, oneof: "message" },
  { no: 44, name: "send_stream_header", kind: "message", T: import_room_pb.SendStreamHeaderRequest, oneof: "message" },
  { no: 45, name: "send_stream_chunk", kind: "message", T: import_room_pb.SendStreamChunkRequest, oneof: "message" },
  { no: 46, name: "send_stream_trailer", kind: "message", T: import_room_pb.SendStreamTrailerRequest, oneof: "message" },
  { no: 47, name: "set_data_channel_buffered_amount_low_threshold", kind: "message", T: import_room_pb.SetDataChannelBufferedAmountLowThresholdRequest, oneof: "message" },
  { no: 49, name: "load_audio_filter_plugin", kind: "message", T: import_audio_frame_pb.LoadAudioFilterPluginRequest, oneof: "message" },
  { no: 50, name: "new_apm", kind: "message", T: import_audio_frame_pb.NewApmRequest, oneof: "message" },
  { no: 51, name: "apm_process_stream", kind: "message", T: import_audio_frame_pb.ApmProcessStreamRequest, oneof: "message" },
  { no: 52, name: "apm_process_reverse_stream", kind: "message", T: import_audio_frame_pb.ApmProcessReverseStreamRequest, oneof: "message" },
  { no: 53, name: "apm_set_stream_delay", kind: "message", T: import_audio_frame_pb.ApmSetStreamDelayRequest, oneof: "message" }
]);
let FfiRequest = _FfiRequest;
const _FfiResponse = class _FfiResponse extends import_protobuf.Message {
  constructor(data) {
    super();
    /**
     * @generated from oneof livekit.proto.FfiResponse.message
     */
    this.message = { case: void 0 };
    import_protobuf.proto2.util.initPartial(data, this);
  }
  static fromBinary(bytes, options) {
    return new _FfiResponse().fromBinary(bytes, options);
  }
  static fromJson(jsonValue, options) {
    return new _FfiResponse().fromJson(jsonValue, options);
  }
  static fromJsonString(jsonString, options) {
    return new _FfiResponse().fromJsonString(jsonString, options);
  }
  static equals(a, b) {
    return import_protobuf.proto2.util.equals(_FfiResponse, a, b);
  }
};
_FfiResponse.runtime = import_protobuf.proto2;
_FfiResponse.typeName = "livekit.proto.FfiResponse";
_FfiResponse.fields = import_protobuf.proto2.util.newFieldList(() => [
  { no: 2, name: "dispose", kind: "message", T: DisposeResponse, oneof: "message" },
  { no: 3, name: "connect", kind: "message", T: import_room_pb.ConnectResponse, oneof: "message" },
  { no: 4, name: "disconnect", kind: "message", T: import_room_pb.DisconnectResponse, oneof: "message" },
  { no: 5, name: "publish_track", kind: "message", T: import_room_pb.PublishTrackResponse, oneof: "message" },
  { no: 6, name: "unpublish_track", kind: "message", T: import_room_pb.UnpublishTrackResponse, oneof: "message" },
  { no: 7, name: "publish_data", kind: "message", T: import_room_pb.PublishDataResponse, oneof: "message" },
  { no: 8, name: "set_subscribed", kind: "message", T: import_room_pb.SetSubscribedResponse, oneof: "message" },
  { no: 9, name: "set_local_metadata", kind: "message", T: import_room_pb.SetLocalMetadataResponse, oneof: "message" },
  { no: 10, name: "set_local_name", kind: "message", T: import_room_pb.SetLocalNameResponse, oneof: "message" },
  { no: 11, name: "set_local_attributes", kind: "message", T: import_room_pb.SetLocalAttributesResponse, oneof: "message" },
  { no: 12, name: "get_session_stats", kind: "message", T: import_room_pb.GetSessionStatsResponse, oneof: "message" },
  { no: 13, name: "publish_transcription", kind: "message", T: import_room_pb.PublishTranscriptionResponse, oneof: "message" },
  { no: 14, name: "publish_sip_dtmf", kind: "message", T: import_room_pb.PublishSipDtmfResponse, oneof: "message" },
  { no: 15, name: "create_video_track", kind: "message", T: import_track_pb.CreateVideoTrackResponse, oneof: "message" },
  { no: 16, name: "create_audio_track", kind: "message", T: import_track_pb.CreateAudioTrackResponse, oneof: "message" },
  { no: 17, name: "local_track_mute", kind: "message", T: import_track_pb.LocalTrackMuteResponse, oneof: "message" },
  { no: 18, name: "enable_remote_track", kind: "message", T: import_track_pb.EnableRemoteTrackResponse, oneof: "message" },
  { no: 19, name: "get_stats", kind: "message", T: import_track_pb.GetStatsResponse, oneof: "message" },
  { no: 47, name: "set_track_subscription_permissions", kind: "message", T: import_track_pb.SetTrackSubscriptionPermissionsResponse, oneof: "message" },
  { no: 20, name: "new_video_stream", kind: "message", T: import_video_frame_pb.NewVideoStreamResponse, oneof: "message" },
  { no: 21, name: "new_video_source", kind: "message", T: import_video_frame_pb.NewVideoSourceResponse, oneof: "message" },
  { no: 22, name: "capture_video_frame", kind: "message", T: import_video_frame_pb.CaptureVideoFrameResponse, oneof: "message" },
  { no: 23, name: "video_convert", kind: "message", T: import_video_frame_pb.VideoConvertResponse, oneof: "message" },
  { no: 24, name: "video_stream_from_participant", kind: "message", T: import_video_frame_pb.VideoStreamFromParticipantResponse, oneof: "message" },
  { no: 25, name: "new_audio_stream", kind: "message", T: import_audio_frame_pb.NewAudioStreamResponse, oneof: "message" },
  { no: 26, name: "new_audio_source", kind: "message", T: import_audio_frame_pb.NewAudioSourceResponse, oneof: "message" },
  { no: 27, name: "capture_audio_frame", kind: "message", T: import_audio_frame_pb.CaptureAudioFrameResponse, oneof: "message" },
  { no: 28, name: "clear_audio_buffer", kind: "message", T: import_audio_frame_pb.ClearAudioBufferResponse, oneof: "message" },
  { no: 29, name: "new_audio_resampler", kind: "message", T: import_audio_frame_pb.NewAudioResamplerResponse, oneof: "message" },
  { no: 30, name: "remix_and_resample", kind: "message", T: import_audio_frame_pb.RemixAndResampleResponse, oneof: "message" },
  { no: 31, name: "audio_stream_from_participant", kind: "message", T: import_audio_frame_pb.AudioStreamFromParticipantResponse, oneof: "message" },
  { no: 32, name: "e2ee", kind: "message", T: import_e2ee_pb.E2eeResponse, oneof: "message" },
  { no: 33, name: "new_sox_resampler", kind: "message", T: import_audio_frame_pb.NewSoxResamplerResponse, oneof: "message" },
  { no: 34, name: "push_sox_resampler", kind: "message", T: import_audio_frame_pb.PushSoxResamplerResponse, oneof: "message" },
  { no: 35, name: "flush_sox_resampler", kind: "message", T: import_audio_frame_pb.FlushSoxResamplerResponse, oneof: "message" },
  { no: 36, name: "send_chat_message", kind: "message", T: import_room_pb.SendChatMessageResponse, oneof: "message" },
  { no: 37, name: "perform_rpc", kind: "message", T: import_rpc_pb.PerformRpcResponse, oneof: "message" },
  { no: 38, name: "register_rpc_method", kind: "message", T: import_rpc_pb.RegisterRpcMethodResponse, oneof: "message" },
  { no: 39, name: "unregister_rpc_method", kind: "message", T: import_rpc_pb.UnregisterRpcMethodResponse, oneof: "message" },
  { no: 40, name: "rpc_method_invocation_response", kind: "message", T: import_rpc_pb.RpcMethodInvocationResponseResponse, oneof: "message" },
  { no: 41, name: "enable_remote_track_publication", kind: "message", T: import_track_publication_pb.EnableRemoteTrackPublicationResponse, oneof: "message" },
  { no: 42, name: "update_remote_track_publication_dimension", kind: "message", T: import_track_publication_pb.UpdateRemoteTrackPublicationDimensionResponse, oneof: "message" },
  { no: 43, name: "send_stream_header", kind: "message", T: import_room_pb.SendStreamHeaderResponse, oneof: "message" },
  { no: 44, name: "send_stream_chunk", kind: "message", T: import_room_pb.SendStreamChunkResponse, oneof: "message" },
  { no: 45, name: "send_stream_trailer", kind: "message", T: import_room_pb.SendStreamTrailerResponse, oneof: "message" },
  { no: 46, name: "set_data_channel_buffered_amount_low_threshold", kind: "message", T: import_room_pb.SetDataChannelBufferedAmountLowThresholdResponse, oneof: "message" },
  { no: 48, name: "load_audio_filter_plugin", kind: "message", T: import_audio_frame_pb.LoadAudioFilterPluginResponse, oneof: "message" },
  { no: 49, name: "new_apm", kind: "message", T: import_audio_frame_pb.NewApmResponse, oneof: "message" },
  { no: 50, name: "apm_process_stream", kind: "message", T: import_audio_frame_pb.ApmProcessStreamResponse, oneof: "message" },
  { no: 51, name: "apm_process_reverse_stream", kind: "message", T: import_audio_frame_pb.ApmProcessReverseStreamResponse, oneof: "message" },
  { no: 52, name: "apm_set_stream_delay", kind: "message", T: import_audio_frame_pb.ApmSetStreamDelayResponse, oneof: "message" }
]);
let FfiResponse = _FfiResponse;
const _FfiEvent = class _FfiEvent extends import_protobuf.Message {
  constructor(data) {
    super();
    /**
     * @generated from oneof livekit.proto.FfiEvent.message
     */
    this.message = { case: void 0 };
    import_protobuf.proto2.util.initPartial(data, this);
  }
  static fromBinary(bytes, options) {
    return new _FfiEvent().fromBinary(bytes, options);
  }
  static fromJson(jsonValue, options) {
    return new _FfiEvent().fromJson(jsonValue, options);
  }
  static fromJsonString(jsonString, options) {
    return new _FfiEvent().fromJsonString(jsonString, options);
  }
  static equals(a, b) {
    return import_protobuf.proto2.util.equals(_FfiEvent, a, b);
  }
};
_FfiEvent.runtime = import_protobuf.proto2;
_FfiEvent.typeName = "livekit.proto.FfiEvent";
_FfiEvent.fields = import_protobuf.proto2.util.newFieldList(() => [
  { no: 1, name: "room_event", kind: "message", T: import_room_pb.RoomEvent, oneof: "message" },
  { no: 2, name: "track_event", kind: "message", T: import_track_pb.TrackEvent, oneof: "message" },
  { no: 3, name: "video_stream_event", kind: "message", T: import_video_frame_pb.VideoStreamEvent, oneof: "message" },
  { no: 4, name: "audio_stream_event", kind: "message", T: import_audio_frame_pb.AudioStreamEvent, oneof: "message" },
  { no: 5, name: "connect", kind: "message", T: import_room_pb.ConnectCallback, oneof: "message" },
  { no: 7, name: "disconnect", kind: "message", T: import_room_pb.DisconnectCallback, oneof: "message" },
  { no: 8, name: "dispose", kind: "message", T: DisposeCallback, oneof: "message" },
  { no: 9, name: "publish_track", kind: "message", T: import_room_pb.PublishTrackCallback, oneof: "message" },
  { no: 10, name: "unpublish_track", kind: "message", T: import_room_pb.UnpublishTrackCallback, oneof: "message" },
  { no: 11, name: "publish_data", kind: "message", T: import_room_pb.PublishDataCallback, oneof: "message" },
  { no: 12, name: "publish_transcription", kind: "message", T: import_room_pb.PublishTranscriptionCallback, oneof: "message" },
  { no: 13, name: "capture_audio_frame", kind: "message", T: import_audio_frame_pb.CaptureAudioFrameCallback, oneof: "message" },
  { no: 14, name: "set_local_metadata", kind: "message", T: import_room_pb.SetLocalMetadataCallback, oneof: "message" },
  { no: 15, name: "set_local_name", kind: "message", T: import_room_pb.SetLocalNameCallback, oneof: "message" },
  { no: 16, name: "set_local_attributes", kind: "message", T: import_room_pb.SetLocalAttributesCallback, oneof: "message" },
  { no: 17, name: "get_stats", kind: "message", T: import_track_pb.GetStatsCallback, oneof: "message" },
  { no: 18, name: "logs", kind: "message", T: LogBatch, oneof: "message" },
  { no: 19, name: "get_session_stats", kind: "message", T: import_room_pb.GetSessionStatsCallback, oneof: "message" },
  { no: 20, name: "panic", kind: "message", T: Panic, oneof: "message" },
  { no: 21, name: "publish_sip_dtmf", kind: "message", T: import_room_pb.PublishSipDtmfCallback, oneof: "message" },
  { no: 22, name: "chat_message", kind: "message", T: import_room_pb.SendChatMessageCallback, oneof: "message" },
  { no: 23, name: "perform_rpc", kind: "message", T: import_rpc_pb.PerformRpcCallback, oneof: "message" },
  { no: 24, name: "rpc_method_invocation", kind: "message", T: import_rpc_pb.RpcMethodInvocationEvent, oneof: "message" },
  { no: 25, name: "send_stream_header", kind: "message", T: import_room_pb.SendStreamHeaderCallback, oneof: "message" },
  { no: 26, name: "send_stream_chunk", kind: "message", T: import_room_pb.SendStreamChunkCallback, oneof: "message" },
  { no: 27, name: "send_stream_trailer", kind: "message", T: import_room_pb.SendStreamTrailerCallback, oneof: "message" }
]);
let FfiEvent = _FfiEvent;
const _DisposeRequest = class _DisposeRequest extends import_protobuf.Message {
  constructor(data) {
    super();
    import_protobuf.proto2.util.initPartial(data, this);
  }
  static fromBinary(bytes, options) {
    return new _DisposeRequest().fromBinary(bytes, options);
  }
  static fromJson(jsonValue, options) {
    return new _DisposeRequest().fromJson(jsonValue, options);
  }
  static fromJsonString(jsonString, options) {
    return new _DisposeRequest().fromJsonString(jsonString, options);
  }
  static equals(a, b) {
    return import_protobuf.proto2.util.equals(_DisposeRequest, a, b);
  }
};
_DisposeRequest.runtime = import_protobuf.proto2;
_DisposeRequest.typeName = "livekit.proto.DisposeRequest";
_DisposeRequest.fields = import_protobuf.proto2.util.newFieldList(() => [
  { no: 1, name: "async", kind: "scalar", T: 8, req: true }
]);
let DisposeRequest = _DisposeRequest;
const _DisposeResponse = class _DisposeResponse extends import_protobuf.Message {
  constructor(data) {
    super();
    import_protobuf.proto2.util.initPartial(data, this);
  }
  static fromBinary(bytes, options) {
    return new _DisposeResponse().fromBinary(bytes, options);
  }
  static fromJson(jsonValue, options) {
    return new _DisposeResponse().fromJson(jsonValue, options);
  }
  static fromJsonString(jsonString, options) {
    return new _DisposeResponse().fromJsonString(jsonString, options);
  }
  static equals(a, b) {
    return import_protobuf.proto2.util.equals(_DisposeResponse, a, b);
  }
};
_DisposeResponse.runtime = import_protobuf.proto2;
_DisposeResponse.typeName = "livekit.proto.DisposeResponse";
_DisposeResponse.fields = import_protobuf.proto2.util.newFieldList(() => [
  { no: 1, name: "async_id", kind: "scalar", T: 4, opt: true }
]);
let DisposeResponse = _DisposeResponse;
const _DisposeCallback = class _DisposeCallback extends import_protobuf.Message {
  constructor(data) {
    super();
    import_protobuf.proto2.util.initPartial(data, this);
  }
  static fromBinary(bytes, options) {
    return new _DisposeCallback().fromBinary(bytes, options);
  }
  static fromJson(jsonValue, options) {
    return new _DisposeCallback().fromJson(jsonValue, options);
  }
  static fromJsonString(jsonString, options) {
    return new _DisposeCallback().fromJsonString(jsonString, options);
  }
  static equals(a, b) {
    return import_protobuf.proto2.util.equals(_DisposeCallback, a, b);
  }
};
_DisposeCallback.runtime = import_protobuf.proto2;
_DisposeCallback.typeName = "livekit.proto.DisposeCallback";
_DisposeCallback.fields = import_protobuf.proto2.util.newFieldList(() => [
  { no: 1, name: "async_id", kind: "scalar", T: 4, req: true }
]);
let DisposeCallback = _DisposeCallback;
const _LogRecord = class _LogRecord extends import_protobuf.Message {
  constructor(data) {
    super();
    import_protobuf.proto2.util.initPartial(data, this);
  }
  static fromBinary(bytes, options) {
    return new _LogRecord().fromBinary(bytes, options);
  }
  static fromJson(jsonValue, options) {
    return new _LogRecord().fromJson(jsonValue, options);
  }
  static fromJsonString(jsonString, options) {
    return new _LogRecord().fromJsonString(jsonString, options);
  }
  static equals(a, b) {
    return import_protobuf.proto2.util.equals(_LogRecord, a, b);
  }
};
_LogRecord.runtime = import_protobuf.proto2;
_LogRecord.typeName = "livekit.proto.LogRecord";
_LogRecord.fields = import_protobuf.proto2.util.newFieldList(() => [
  { no: 1, name: "level", kind: "enum", T: import_protobuf.proto2.getEnumType(LogLevel), req: true },
  { no: 2, name: "target", kind: "scalar", T: 9, req: true },
  { no: 3, name: "module_path", kind: "scalar", T: 9, opt: true },
  { no: 4, name: "file", kind: "scalar", T: 9, opt: true },
  { no: 5, name: "line", kind: "scalar", T: 13, opt: true },
  { no: 6, name: "message", kind: "scalar", T: 9, req: true }
]);
let LogRecord = _LogRecord;
const _LogBatch = class _LogBatch extends import_protobuf.Message {
  constructor(data) {
    super();
    /**
     * @generated from field: repeated livekit.proto.LogRecord records = 1;
     */
    this.records = [];
    import_protobuf.proto2.util.initPartial(data, this);
  }
  static fromBinary(bytes, options) {
    return new _LogBatch().fromBinary(bytes, options);
  }
  static fromJson(jsonValue, options) {
    return new _LogBatch().fromJson(jsonValue, options);
  }
  static fromJsonString(jsonString, options) {
    return new _LogBatch().fromJsonString(jsonString, options);
  }
  static equals(a, b) {
    return import_protobuf.proto2.util.equals(_LogBatch, a, b);
  }
};
_LogBatch.runtime = import_protobuf.proto2;
_LogBatch.typeName = "livekit.proto.LogBatch";
_LogBatch.fields = import_protobuf.proto2.util.newFieldList(() => [
  { no: 1, name: "records", kind: "message", T: LogRecord, repeated: true }
]);
let LogBatch = _LogBatch;
const _Panic = class _Panic extends import_protobuf.Message {
  constructor(data) {
    super();
    import_protobuf.proto2.util.initPartial(data, this);
  }
  static fromBinary(bytes, options) {
    return new _Panic().fromBinary(bytes, options);
  }
  static fromJson(jsonValue, options) {
    return new _Panic().fromJson(jsonValue, options);
  }
  static fromJsonString(jsonString, options) {
    return new _Panic().fromJsonString(jsonString, options);
  }
  static equals(a, b) {
    return import_protobuf.proto2.util.equals(_Panic, a, b);
  }
};
_Panic.runtime = import_protobuf.proto2;
_Panic.typeName = "livekit.proto.Panic";
_Panic.fields = import_protobuf.proto2.util.newFieldList(() => [
  { no: 1, name: "message", kind: "scalar", T: 9, req: true }
]);
let Panic = _Panic;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  DisposeCallback,
  DisposeRequest,
  DisposeResponse,
  FfiEvent,
  FfiRequest,
  FfiResponse,
  LogBatch,
  LogLevel,
  LogRecord,
  Panic
});
