// @livekit/rtc-node@0.13.14/dist/proto/participant_pb.js
import { Message, proto2 } from "@bufbuild/protobuf";
import { FfiOwnedHandle } from "./handle_pb.js";
var ParticipantKind = /* @__PURE__ */ ((ParticipantKind2) => {
  ParticipantKind2[ParticipantKind2["STANDARD"] = 0] = "STANDARD";
  ParticipantKind2[ParticipantKind2["INGRESS"] = 1] = "INGRESS";
  ParticipantKind2[ParticipantKind2["EGRESS"] = 2] = "EGRESS";
  ParticipantKind2[ParticipantKind2["SIP"] = 3] = "SIP";
  ParticipantKind2[ParticipantKind2["AGENT"] = 4] = "AGENT";
  return ParticipantKind2;
})(ParticipantKind || {});
proto2.util.setEnumType(ParticipantKind, "livekit.proto.ParticipantKind", [
  { no: 0, name: "PARTICIPANT_KIND_STANDARD" },
  { no: 1, name: "PARTICIPANT_KIND_INGRESS" },
  { no: 2, name: "PARTICIPANT_KIND_EGRESS" },
  { no: 3, name: "PARTICIPANT_KIND_SIP" },
  { no: 4, name: "PARTICIPANT_KIND_AGENT" }
]);
var DisconnectReason = /* @__PURE__ */ ((DisconnectReason2) => {
  DisconnectReason2[DisconnectReason2["UNKNOWN_REASON"] = 0] = "UNKNOWN_REASON";
  DisconnectReason2[DisconnectReason2["CLIENT_INITIATED"] = 1] = "CLIENT_INITIATED";
  DisconnectReason2[DisconnectReason2["DUPLICATE_IDENTITY"] = 2] = "DUPLICATE_IDENTITY";
  DisconnectReason2[DisconnectReason2["SERVER_SHUTDOWN"] = 3] = "SERVER_SHUTDOWN";
  DisconnectReason2[DisconnectReason2["PARTICIPANT_REMOVED"] = 4] = "PARTICIPANT_REMOVED";
  DisconnectReason2[DisconnectReason2["ROOM_DELETED"] = 5] = "ROOM_DELETED";
  DisconnectReason2[DisconnectReason2["STATE_MISMATCH"] = 6] = "STATE_MISMATCH";
  DisconnectReason2[DisconnectReason2["JOIN_FAILURE"] = 7] = "JOIN_FAILURE";
  DisconnectReason2[DisconnectReason2["MIGRATION"] = 8] = "MIGRATION";
  DisconnectReason2[DisconnectReason2["SIGNAL_CLOSE"] = 9] = "SIGNAL_CLOSE";
  DisconnectReason2[DisconnectReason2["ROOM_CLOSED"] = 10] = "ROOM_CLOSED";
  DisconnectReason2[DisconnectReason2["USER_UNAVAILABLE"] = 11] = "USER_UNAVAILABLE";
  DisconnectReason2[DisconnectReason2["USER_REJECTED"] = 12] = "USER_REJECTED";
  DisconnectReason2[DisconnectReason2["SIP_TRUNK_FAILURE"] = 13] = "SIP_TRUNK_FAILURE";
  return DisconnectReason2;
})(DisconnectReason || {});
proto2.util.setEnumType(DisconnectReason, "livekit.proto.DisconnectReason", [
  { no: 0, name: "UNKNOWN_REASON" },
  { no: 1, name: "CLIENT_INITIATED" },
  { no: 2, name: "DUPLICATE_IDENTITY" },
  { no: 3, name: "SERVER_SHUTDOWN" },
  { no: 4, name: "PARTICIPANT_REMOVED" },
  { no: 5, name: "ROOM_DELETED" },
  { no: 6, name: "STATE_MISMATCH" },
  { no: 7, name: "JOIN_FAILURE" },
  { no: 8, name: "MIGRATION" },
  { no: 9, name: "SIGNAL_CLOSE" },
  { no: 10, name: "ROOM_CLOSED" },
  { no: 11, name: "USER_UNAVAILABLE" },
  { no: 12, name: "USER_REJECTED" },
  { no: 13, name: "SIP_TRUNK_FAILURE" }
]);
const _ParticipantInfo = class _ParticipantInfo extends Message {
  constructor(data) {
    super();
    /**
     * @generated from field: map<string, string> attributes = 5;
     */
    this.attributes = {};
    proto2.util.initPartial(data, this);
  }
  static fromBinary(bytes, options) {
    return new _ParticipantInfo().fromBinary(bytes, options);
  }
  static fromJson(jsonValue, options) {
    return new _ParticipantInfo().fromJson(jsonValue, options);
  }
  static fromJsonString(jsonString, options) {
    return new _ParticipantInfo().fromJsonString(jsonString, options);
  }
  static equals(a, b) {
    return proto2.util.equals(_ParticipantInfo, a, b);
  }
};
_ParticipantInfo.runtime = proto2;
_ParticipantInfo.typeName = "livekit.proto.ParticipantInfo";
_ParticipantInfo.fields = proto2.util.newFieldList(() => [
  { no: 1, name: "sid", kind: "scalar", T: 9, req: true },
  { no: 2, name: "name", kind: "scalar", T: 9, req: true },
  { no: 3, name: "identity", kind: "scalar", T: 9, req: true },
  { no: 4, name: "metadata", kind: "scalar", T: 9, req: true },
  { no: 5, name: "attributes", kind: "map", K: 9, V: {
    kind: "scalar",
    T: 9
    /* ScalarType.STRING */
  } },
  { no: 6, name: "kind", kind: "enum", T: proto2.getEnumType(ParticipantKind), req: true },
  { no: 7, name: "disconnect_reason", kind: "enum", T: proto2.getEnumType(DisconnectReason), req: true }
]);
let ParticipantInfo = _ParticipantInfo;
const _OwnedParticipant = class _OwnedParticipant extends Message {
  constructor(data) {
    super();
    proto2.util.initPartial(data, this);
  }
  static fromBinary(bytes, options) {
    return new _OwnedParticipant().fromBinary(bytes, options);
  }
  static fromJson(jsonValue, options) {
    return new _OwnedParticipant().fromJson(jsonValue, options);
  }
  static fromJsonString(jsonString, options) {
    return new _OwnedParticipant().fromJsonString(jsonString, options);
  }
  static equals(a, b) {
    return proto2.util.equals(_OwnedParticipant, a, b);
  }
};
_OwnedParticipant.runtime = proto2;
_OwnedParticipant.typeName = "livekit.proto.OwnedParticipant";
_OwnedParticipant.fields = proto2.util.newFieldList(() => [
  { no: 1, name: "handle", kind: "message", T: FfiOwnedHandle, req: true },
  { no: 2, name: "info", kind: "message", T: ParticipantInfo, req: true }
]);
let OwnedParticipant = _OwnedParticipant;
export {
  DisconnectReason,
  OwnedParticipant,
  ParticipantInfo,
  ParticipantKind
};
