// @livekit/rtc-node@0.13.14/dist/proto/participant_pb.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var participant_pb_exports = {};
__export(participant_pb_exports, {
  DisconnectReason: () => DisconnectReason,
  OwnedParticipant: () => OwnedParticipant,
  ParticipantInfo: () => ParticipantInfo,
  ParticipantKind: () => ParticipantKind
});
module.exports = __toCommonJS(participant_pb_exports);
var import_protobuf = require("@bufbuild/protobuf");
var import_handle_pb = require("./handle_pb.cjs");
var ParticipantKind = /* @__PURE__ */ ((ParticipantKind2) => {
  ParticipantKind2[ParticipantKind2["STANDARD"] = 0] = "STANDARD";
  ParticipantKind2[ParticipantKind2["INGRESS"] = 1] = "INGRESS";
  ParticipantKind2[ParticipantKind2["EGRESS"] = 2] = "EGRESS";
  ParticipantKind2[ParticipantKind2["SIP"] = 3] = "SIP";
  ParticipantKind2[ParticipantKind2["AGENT"] = 4] = "AGENT";
  return ParticipantKind2;
})(ParticipantKind || {});
import_protobuf.proto2.util.setEnumType(ParticipantKind, "livekit.proto.ParticipantKind", [
  { no: 0, name: "PARTICIPANT_KIND_STANDARD" },
  { no: 1, name: "PARTICIPANT_KIND_INGRESS" },
  { no: 2, name: "PARTICIPANT_KIND_EGRESS" },
  { no: 3, name: "PARTICIPANT_KIND_SIP" },
  { no: 4, name: "PARTICIPANT_KIND_AGENT" }
]);
var DisconnectReason = /* @__PURE__ */ ((DisconnectReason2) => {
  DisconnectReason2[DisconnectReason2["UNKNOWN_REASON"] = 0] = "UNKNOWN_REASON";
  DisconnectReason2[DisconnectReason2["CLIENT_INITIATED"] = 1] = "CLIENT_INITIATED";
  DisconnectReason2[DisconnectReason2["DUPLICATE_IDENTITY"] = 2] = "DUPLICATE_IDENTITY";
  DisconnectReason2[DisconnectReason2["SERVER_SHUTDOWN"] = 3] = "SERVER_SHUTDOWN";
  DisconnectReason2[DisconnectReason2["PARTICIPANT_REMOVED"] = 4] = "PARTICIPANT_REMOVED";
  DisconnectReason2[DisconnectReason2["ROOM_DELETED"] = 5] = "ROOM_DELETED";
  DisconnectReason2[DisconnectReason2["STATE_MISMATCH"] = 6] = "STATE_MISMATCH";
  DisconnectReason2[DisconnectReason2["JOIN_FAILURE"] = 7] = "JOIN_FAILURE";
  DisconnectReason2[DisconnectReason2["MIGRATION"] = 8] = "MIGRATION";
  DisconnectReason2[DisconnectReason2["SIGNAL_CLOSE"] = 9] = "SIGNAL_CLOSE";
  DisconnectReason2[DisconnectReason2["ROOM_CLOSED"] = 10] = "ROOM_CLOSED";
  DisconnectReason2[DisconnectReason2["USER_UNAVAILABLE"] = 11] = "USER_UNAVAILABLE";
  DisconnectReason2[DisconnectReason2["USER_REJECTED"] = 12] = "USER_REJECTED";
  DisconnectReason2[DisconnectReason2["SIP_TRUNK_FAILURE"] = 13] = "SIP_TRUNK_FAILURE";
  return DisconnectReason2;
})(DisconnectReason || {});
import_protobuf.proto2.util.setEnumType(DisconnectReason, "livekit.proto.DisconnectReason", [
  { no: 0, name: "UNKNOWN_REASON" },
  { no: 1, name: "CLIENT_INITIATED" },
  { no: 2, name: "DUPLICATE_IDENTITY" },
  { no: 3, name: "SERVER_SHUTDOWN" },
  { no: 4, name: "PARTICIPANT_REMOVED" },
  { no: 5, name: "ROOM_DELETED" },
  { no: 6, name: "STATE_MISMATCH" },
  { no: 7, name: "JOIN_FAILURE" },
  { no: 8, name: "MIGRATION" },
  { no: 9, name: "SIGNAL_CLOSE" },
  { no: 10, name: "ROOM_CLOSED" },
  { no: 11, name: "USER_UNAVAILABLE" },
  { no: 12, name: "USER_REJECTED" },
  { no: 13, name: "SIP_TRUNK_FAILURE" }
]);
const _ParticipantInfo = class _ParticipantInfo extends import_protobuf.Message {
  constructor(data) {
    super();
    /**
     * @generated from field: map<string, string> attributes = 5;
     */
    this.attributes = {};
    import_protobuf.proto2.util.initPartial(data, this);
  }
  static fromBinary(bytes, options) {
    return new _ParticipantInfo().fromBinary(bytes, options);
  }
  static fromJson(jsonValue, options) {
    return new _ParticipantInfo().fromJson(jsonValue, options);
  }
  static fromJsonString(jsonString, options) {
    return new _ParticipantInfo().fromJsonString(jsonString, options);
  }
  static equals(a, b) {
    return import_protobuf.proto2.util.equals(_ParticipantInfo, a, b);
  }
};
_ParticipantInfo.runtime = import_protobuf.proto2;
_ParticipantInfo.typeName = "livekit.proto.ParticipantInfo";
_ParticipantInfo.fields = import_protobuf.proto2.util.newFieldList(() => [
  { no: 1, name: "sid", kind: "scalar", T: 9, req: true },
  { no: 2, name: "name", kind: "scalar", T: 9, req: true },
  { no: 3, name: "identity", kind: "scalar", T: 9, req: true },
  { no: 4, name: "metadata", kind: "scalar", T: 9, req: true },
  { no: 5, name: "attributes", kind: "map", K: 9, V: {
    kind: "scalar",
    T: 9
    /* ScalarType.STRING */
  } },
  { no: 6, name: "kind", kind: "enum", T: import_protobuf.proto2.getEnumType(ParticipantKind), req: true },
  { no: 7, name: "disconnect_reason", kind: "enum", T: import_protobuf.proto2.getEnumType(DisconnectReason), req: true }
]);
let ParticipantInfo = _ParticipantInfo;
const _OwnedParticipant = class _OwnedParticipant extends import_protobuf.Message {
  constructor(data) {
    super();
    import_protobuf.proto2.util.initPartial(data, this);
  }
  static fromBinary(bytes, options) {
    return new _OwnedParticipant().fromBinary(bytes, options);
  }
  static fromJson(jsonValue, options) {
    return new _OwnedParticipant().fromJson(jsonValue, options);
  }
  static fromJsonString(jsonString, options) {
    return new _OwnedParticipant().fromJsonString(jsonString, options);
  }
  static equals(a, b) {
    return import_protobuf.proto2.util.equals(_OwnedParticipant, a, b);
  }
};
_OwnedParticipant.runtime = import_protobuf.proto2;
_OwnedParticipant.typeName = "livekit.proto.OwnedParticipant";
_OwnedParticipant.fields = import_protobuf.proto2.util.newFieldList(() => [
  { no: 1, name: "handle", kind: "message", T: import_handle_pb.FfiOwnedHandle, req: true },
  { no: 2, name: "info", kind: "message", T: ParticipantInfo, req: true }
]);
let OwnedParticipant = _OwnedParticipant;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  DisconnectReason,
  OwnedParticipant,
  ParticipantInfo,
  ParticipantKind
});
