// @livekit/rtc-node@0.13.14/dist/data_streams/stream_reader.js
import { log } from "../log.js";
import { bigIntToNumber } from "../utils.js";
class BaseStreamReader {
  get info() {
    return this._info;
  }
  constructor(info, stream, totalByteSize) {
    this.reader = stream;
    this.totalByteSize = totalByteSize;
    this._info = info;
    this.bytesReceived = 0;
  }
}
class ByteStreamReader extends BaseStreamReader {
  handleChunkReceived(chunk) {
    var _a;
    this.bytesReceived += chunk.content.byteLength;
    const currentProgress = this.totalByteSize ? this.bytesReceived / this.totalByteSize : void 0;
    (_a = this.onProgress) == null ? void 0 : _a.call(this, currentProgress);
  }
  [Symbol.asyncIterator]() {
    const reader = this.reader.getReader();
    return {
      next: async () => {
        try {
          const { done, value } = await reader.read();
          if (done) {
            return { done: true, value: void 0 };
          } else {
            this.handleChunkReceived(value);
            return { done: false, value: value.content };
          }
        } catch (error) {
          log.error("error processing stream update", error);
          return { done: true, value: void 0 };
        }
      },
      return() {
        reader.releaseLock();
        return { done: true, value: void 0 };
      }
    };
  }
  async readAll() {
    const chunks = /* @__PURE__ */ new Set();
    for await (const chunk of this) {
      chunks.add(chunk);
    }
    return Array.from(chunks);
  }
}
class TextStreamReader extends BaseStreamReader {
  /**
   * A TextStreamReader instance can be used as an AsyncIterator that returns the entire string
   * that has been received up to the current point in time.
   */
  constructor(info, stream, totalChunkCount) {
    super(info, stream, totalChunkCount);
    this.receivedChunks = /* @__PURE__ */ new Map();
  }
  handleChunkReceived(chunk) {
    var _a;
    const index = bigIntToNumber(chunk.chunkIndex);
    const previousChunkAtIndex = this.receivedChunks.get(index);
    if (previousChunkAtIndex && previousChunkAtIndex.version > chunk.version) {
      return;
    }
    this.receivedChunks.set(index, chunk);
    const currentProgress = this.totalByteSize ? this.receivedChunks.size / this.totalByteSize : void 0;
    (_a = this.onProgress) == null ? void 0 : _a.call(this, currentProgress);
  }
  /**
   * Async iterator implementation to allow usage of `for await...of` syntax.
   * Yields structured chunks from the stream.
   *
   */
  [Symbol.asyncIterator]() {
    const reader = this.reader.getReader();
    const decoder = new TextDecoder();
    return {
      next: async () => {
        try {
          const { done, value } = await reader.read();
          if (done) {
            return { done: true, value: void 0 };
          } else {
            this.handleChunkReceived(value);
            return {
              done: false,
              value: decoder.decode(value.content)
            };
          }
        } catch (error) {
          log.error("error processing stream update", error);
          return { done: true, value: void 0 };
        }
      },
      return() {
        reader.releaseLock();
        return { done: true, value: void 0 };
      }
    };
  }
  async readAll() {
    let finalString = "";
    for await (const chunk of this) {
      finalString += chunk;
    }
    return finalString;
  }
}
export {
  ByteStreamReader,
  TextStreamReader
};
